clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/model/AzureServerGroupDescription.groovy

```diff
@@ -50,13 +53,77 @@ class AzureServerGroupDescription extends AzureResourceOpsDescription {
   static class AzureOperatingSystemConfig {
     String adminUserName
     String adminPassword
+    String computerNamePrefix
   }
 
   Integer getStorageAccountCount() {
     (sku.capacity / 20) + 1
   }
 
+  static UpgradePolicy getPolicyFromMode(String mode) {
+    mode.toLowerCase() == "Automatic".toLowerCase() ? UpgradePolicy.Automatic : UpgradePolicy.Manual
+  }
+
+  // For now, same thing as identifier
+  String getClusterName() {
+    getIdentifier()
+  }
+
   String getIdentifier() {
     String.format("%s-%s-%s", application, stack, detail)
   }
+
+  static AzureServerGroupDescription build(VirtualMachineScaleSet scaleSet) {
+    def azureSG = new AzureServerGroupDescription()
+    azureSG.name = scaleSet.name
+    azureSG.appName = AzureUtilities.getAppNameFromResourceId(scaleSet.id)
+    // Get the values from the tags if they exist
+    azureSG.tags = scaleSet.tags ? scaleSet.tags : [:]
+    azureSG.stack = scaleSet.tags["stack"] ? scaleSet.tags["stack"] : ""
+    azureSG.detail = scaleSet.tags["detail"] ? scaleSet.tags["detail"] : ""
+    azureSG.application = scaleSet.tags["appName"] ? scaleSet.tags["appName"] : azureSG.appName
+    azureSG.clusterName = scaleSet.tags["cluster"] ?
+      scaleSet.tags["cluster"] : azureSG.getClusterName()
+
+    azureSG.region = scaleSet.location
+
+    azureSG.upgradePolicy = getPolicyFromMode(scaleSet.upgradePolicy.mode)
+
+    // Get the image reference data
+    def image = new AzureNamedImage()
+    def imgRef = scaleSet.virtualMachineProfile?.storageProfile?.imageReference
+    if (imgRef) {
+      image.offer = imgRef.offer
+      image.publisher = imgRef.publisher
+      image.sku = imgRef.sku
+      image.version = imgRef.version
+    }
```

Comment:
I've created a work item to handle this and will tackle that when we build out the rest of the server group data (e.g., associated load balancer, instances)